### PR TITLE
Fix attachments

### PR DESCRIPTION
**Pull request by hbcarlos**
Fixes Markdown cell's attachments and updates migration guide.

## Code changes
Syncs attachments with the shared model.

## User-facing changes
N/A

## Backwards-incompatible changes
N/A

### CODE REVIEW

File: packages/cells/src/model.ts [hunk 1]
```diff
@@ -412,13 +429,52 @@ export abstract class AttachmentsCellModel extends CellModel {
...
+  /**
+   * Dispose of the resources held by the model.
+   */
+  dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+    this._attachments.stateChanged.disconnect(this.onGenericChange, this);
+    this._attachments.changed.disconnect(this._onAttachmentsChange, this);
+    this.sharedModel.changed.disconnect(this._onSharedModelChanged, this);
```
Comment (fcollonval) [inline packages/cells/src/model.ts:441]: The attachments container does not seem to be disposed:

```suggestion
    this._attachements.dispose();
    this.sharedModel.changed.disconnect(this._onSharedModelChanged, this);
```

File: docs/source/extension/extension_migration.rst [hunk 2]
```diff
@@ -30,6 +30,8 @@ bumped their major version (following semver convention). We want to point out p
    See https://github.com/jupyterlab/jupyterlab/pull/11537 for more details.
 - ``@jupyterlab/apputils`` from 3.x to 4.x
    Rename ``IToolbarWidgetRegistry.registerFactory`` to ``IToolbarWidgetRegistry.addFactory``
+- ``@jupyterlab/attachments`` from 3.x to 4.x
+   Removed ``modelDB`` from ``IAttachmentsModel.IOptions``.
```
[the remark this answers is not in the document]
Comment (hbcarlos) [reply docs/source/extension/extension_migration.rst:34]: We added the shared model in v4.
See v3.6:
https://github.com/jupyterlab/jupyterlab/blob/cf523892b4b4d4d7b29bc15f20506a29c347f7df/packages/attachments/src/model.ts#L97-L114

File: packages/cells/src/model.ts [hunk 3]
```diff
@@ -412,13 +429,53 @@ export abstract class AttachmentsCellModel extends CellModel {
...
+  /**
+   * Dispose of the resources held by the model.
+   */
+  dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+    this._attachments.stateChanged.disconnect(this.onGenericChange, this);
+    this._attachments.changed.disconnect(this._onAttachmentsChange, this);
+    this._attachements.dispose();
```
Comment (fcollonval) [inline packages/cells/src/model.ts:441]:
```suggestion
    this._attachments.dispose();
```

**Review by fcollonval — APPROVED** [review]
Thanks @hbcarlos

**Pull request merged**